app/models/skin_parent.rb

```diff
@@ -10,9 +10,9 @@ class SkinParent < ApplicationRecord
 
   validate :no_site_parent
   def no_site_parent
-    if parent_skin.get_role == "site" && !%w(override site).include?(child_skin.get_role)
-      errors.add(:base, ts("^You can't use %{title} as a parent unless replacing the default archive skin.", title: parent_skin.title))
-    end
+    return unless parent_skin&.get_role == "site" && !%w[override site].include?(child_skin.get_role)
+
+    errors.add(:base, ts("^You can't use %{title} as a parent unless replacing the default archive skin.", title: parent_skin.title))
```

Comment:
I can't remember, but maybe someone else can: do we still need `^` here?

Reply:
I've tried removing it and everything looks OK —will remove it.
![image](https://github.com/otwcode/otwarchive/assets/80141759/765df4cf-f4a1-4dc9-89a5-100ee583007c)

Reply:
I think we do need it, judging from that screen shot -- if we take it out, "Skin parent base" is added before the desired text. I feel like there's another way to keep that from happening but I'm blanking on what it is.

Reply:
You should be able to suppress that prefix using the locales file by setting `format: "%{message}"` in the appropriate `activerecord.errors.models` translation scope. (Specifically, I think you need to set it for `skin_parent` and `skin/skin_parents` — the related work/parent work relationships stuff in `config/locales/models/en.yml` is probably a good model to follow in this case.)

It applies to all error messages, so if you want to suppress the prefix that way, it'd be a good idea to go through all of the validations in the `SkinParent` class to make sure that the error messages are still properly formatted. Alternatively, you may be able to apply it to just `base` messages.

Reply:
I've tried adding this to `en.activerecord.errors.models` and using  `errors.add(:base, :site_parent, title: parent_skin.title)`  but the prefix continues to show up:
```
        skin:
          attributes:
            skin_parents:
              base:
                site_parent: You can't use %{title} as a parent unless replacing the default archive skin.
          format: "%{message}"
        skin_parent:
          attributes:
            base:
              site_parent: You can't use %{title} as a parent unless replacing the default archive skin.
          format: "%{message}"
 ```
 
 Is there something I am doing incorrectly?

Reply:
You need to set `format: %{message}` for `skin/skin_parents`, not for `skin`. In the translation code, `/` isn't used to mean nesting; it's just another character that can be used in a key. Like with the `work/parent_work_relationships` key: https://github.com/otwcode/otwarchive/blob/ba94c04580de2bac4c7a932e33a11fad4de7e06d/config/locales/models/en.yml#L153-L154
The class of the parent object is `Work`. The name of the association is `parent_work_relationships`. So when the related work error is displayed on the work, it uses the `work/parent_work_relationships` key to determine the formatting. Similarly, when the skin parent error is displayed on the skin, it uses the `skin/skin_parents` key to determine the formatting.

Reply:
Thank you for expalining, I was trying to Google this but din't know how or where to look.

Reply:
I used to have trouble finding resources on how strings are generated with i18n (aside from what's covered in the [Rails guide](https://guides.rubyonrails.org/i18n.html)). So I ended up adding a monkeypatch that prints out all i18n key look-ups, inspired by [the instructions here](http://www.unixgods.org/Rails/where_is_Rails_trying_to_lookup_L10N_strings.html) but starting with [the current version of the lookup function](https://github.com/ruby-i18n/i18n/blob/7cf09474b77fd41e65d979134b0525f67cf371b0/lib/i18n/backend/simple.rb#L88-L107). It can produce a LOT of spam in the logs when it's enabled, but I can usually track down the information I'm looking for eventually.

Have you double-checked that the error messages look okay for the other validations? Switching to the `%{message}` format often breaks the formatting for errors produced by the built-in validators like `presence` and `uniqueness`, because those error messages are designed around the default `%{attribute} %{message}` error formatting.

Reply:
Thanks for the tip! It slipped my mind to check the other messages, I will do it soon.